source/linux/network.c

```diff
@@ -359,6 +359,7 @@ int get_network_config_and_transfer(struct aws_iotdevice_network_ifconfig *ifcon
         AWS_ZERO_STRUCT(ifr);
 
         strncpy(ifr.ifr_name, address->ifa_name, IFNAMSIZ);
+        ifr.ifr_name[IFNAMSIZ - 1] = 0;
```

Comment:
```
struct aws_iotdevice_network_iface {
    struct aws_allocator *allocator;
    char iface_name[IFACE_NAME_SIZE];
    char ipv4_addr_str[IPV4_ADDRESS_SIZE];
    struct aws_iotdevice_metric_network_transfer metrics;
};
```

This is more for my understanding of the coding style. I remember people usually define the array with +1 size. For example:
```
char iface_name[IFACE_NAME_SIZE + 1];
```

Then the strncpy becomes:
```
strncpy(ifr.ifr_name, address->ifa_name, sizeof(ifr.ifr_name));
```

Is there a reason not to use +1 sizing when defining an array?

Reply:
I think it makes sense that the iface_name is 16 bytes. I believe it is expected that the interface name is at most 16 bytes (15 visible chars and a trailing null). The problem is just that strncpy does not append the null char in the case where the source >= the bound. But setting the last byte to null after the strncpy appeases the compiler warning.

Reply:
strncpy_s() may help clean up the flow a bit but I think this works to cover that edge case as well.

`size_t chars_to_copy = strlen(address->ifa_name) < IFACE_NAME_SIZE ? strlen(address->ifa_name) : IFACE_NAME_SIZE;`
`if (strncpy_s(ifr.ifr_name, IFNAMSIZ, address->ifa_name, chars_to_copy )) { /* this is an error on the copy */ }`